js/adapt-xapi.js

```diff
@@ -416,8 +416,9 @@ define(function(require) {
       if (!statement) {
         return;
       }
-
+      Adapt.trigger("xapi:willSendStatement", statement);
```

Comment:
Also, on that branch you'll find an equivalent call to sendStatements which will need similar events.